File: core/src/test/java/com/larryhsiao/nyx/jots/JotUriTest.java
package com.larryhsiao.nyx.jots;

import com.larryhsiao.nyx.core.jots.ConstJot;
import com.larryhsiao.nyx.core.jots.JotUri;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;


/**
 * Unit-test for the class {@link JotUri}
 */
public class JotUriTest {
    /**
     * Check simple output.
     */
    @Test
    public void simple() {
        assertEquals(
            "http://localhost.com/jots/1",
            new JotUri(
                "http://localhost.com/",
                new ConstJot(
                    1,
                    "content",
                    0,
                    new double[]{},
                    "",
                    1,
                    false)
            ).value().toASCIIString()
        );
    }
}

File: core/src/test/java/com/larryhsiao/nyx/core/jots/JotUriTest.kt
package com.larryhsiao.nyx.core.jots

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Unit-test for the class [JotUri]
 */
class JotUriTest {
    /**
     * Check simple output.
     */
    @Test
    fun simple() {
        Assertions.assertEquals(
            "http://localhost.com/jots/1",
            JotUri(
                "http://localhost.com/",
                ConstJot(
                    1,
                    "",
                    "content",
                    0, doubleArrayOf(),
                    "",
                    1,
                    false)
            ).value().toASCIIString()
        )
    }
}